x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BasicBlockTests.java

```diff
@@ -26,51 +36,57 @@
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class BasicBlockTests extends ESTestCase {
 
-    public void testEmpty() {
-        assertThat(
-            new IntArrayBlock(new int[] {}, 0, new int[] {}, new BitSet(), randomFrom(Block.MvOrdering.values())).getPositionCount(),
-            is(0)
-        );
-        assertThat(IntBlock.newBlockBuilder(0).build().getPositionCount(), is(0));
-        assertThat(new IntArrayVector(new int[] {}, 0).getPositionCount(), is(0));
-        assertThat(IntVector.newVectorBuilder(0).build().getPositionCount(), is(0));
+    final CircuitBreaker breaker = new CountingCircuitBreaker("ESQL-test-breaker");
+    final BlockFactory blockFactory = BlockFactory.getInstance(breaker, bigArrays());
 
-        assertThat(
-            new LongArrayBlock(new long[] {}, 0, new int[] {}, new BitSet(), randomFrom(Block.MvOrdering.values())).getPositionCount(),
-            is(0)
-        );
-        assertThat(LongBlock.newBlockBuilder(0).build().getPositionCount(), is(0));
-        assertThat(new LongArrayVector(new long[] {}, 0).getPositionCount(), is(0));
-        assertThat(LongVector.newVectorBuilder(0).build().getPositionCount(), is(0));
+    BigArrays bigArrays() {
+        var breakerService = mock(CircuitBreakerService.class);
+        when(breakerService.getBreaker(CircuitBreaker.REQUEST)).thenReturn(breaker);
+        return new MockBigArrays(PageCacheRecycler.NON_RECYCLING_INSTANCE, breakerService);
```

Comment:
The `MockBigArrays` ctor that takes a `ByteSizeValue` gets you something pretty similar to this. There's a limit, but you can set it to 1GB and mostly ignore it.

Reply:
ah, that might be more straightforward. I'll update.

Reply:
I did not use MockBigArrays, as it creates a new instance of the limit breaker for each getBreaker on the breaker service call.  This seems like it could be a bug in MockBigArrays?  Well... it depends on the expected usage.

Reply:
> This seems like it could be a bug in MockBigArrays?

I think that's a bug, yeah. But not one to fix now.